tokio/src/sync/mpsc/unbounded.rs

```diff
@@ -172,6 +172,56 @@ impl<T> UnboundedReceiver<T> {
         poll_fn(|cx| self.poll_recv(cx)).await
     }
 
+    /// Receives the next values for this receiver and extends `buffer`.
+    ///
+    /// This method returns the number of values populated in `buffer`.
+    ///
+    /// This method returns 0 if the channel has been closed and there are
+    /// no remaining messages in the channel's queue. This indicates that no
+    /// further values can ever be received from this `Receiver`. The channel is
+    /// closed when all senders have been dropped, or when [`close`] is called.
+    ///
+    /// If there are no messages in the channel's queue, but the channel has
+    /// not yet been closed, this method will sleep until a message is sent or
+    /// the channel is closed.
+    ///
+    /// If at the time of the call `buffer` has no unused capacity,
+    /// additional elements are reserved. Otherwise `recv_many` populates
+    /// `buffer` with no more elements than its unused capacity.
```

Comment:
Can we reverse this sentence? I think the case where `len < capacity` is more common, so I'd like to explain it first.

Reply:
Yes.  I like:

_If at the time of the call `buffer` has unused capacity, `recv_many` extends the buffer with no more elements than
its unused capacity. Should `buffer` have no initial unused capacity, additional elements are first reserved._